overlord/snapstate/snapstate.go

```diff
@@ -131,8 +131,21 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 
 		if experimentalRefreshAppAwareness {
 			if err := SoftNothingRunningRefreshCheck(info); err != nil {
-				// TODO Remember the inhibition time in snap state.
-				return nil, err
+				var now = time.Now()
+				// TODO: find a better location for this amount.
+				const maxInhibition = time.Hour * 24 * 7
```

Comment:
it should probably go close to maxPostponement in autorefresh.go, there is an argument to be made that the code in this block should be a helper there  `inhibitRefresh`